src/layer/rail.js

```diff
@@ -0,0 +1,672 @@
+"use strict";
+
+import * as Color from "../constants/color.js";
+import * as Util from "../js/util.js";
+
+// Exponent base for inter-zoom interpolation
+let railExp = 1.2;
+
+// Helper function to create a "filter" block for a particular railway class.
+function filterRail(railClass, service, brunnel) {
+  return [
+    "all",
+    brunnel === "surface"
+      ? ["!in", "brunnel", "bridge", "tunnel"]
+      : ["==", "brunnel", brunnel],
+    ["in", "class", "rail", "transit"],
+    ["==", "subclass", railClass],
+    [service ? "in" : "!in", "service", "siding", "spur", "yard"],
+  ];
+}
+
+// Base definition that applies to all railways
+var defRail = {
+  type: "line",
+  source: "openmaptiles",
+  "source-layer": "transportation",
+};
+
+var serviceSelector = ["match", ["get", "service"], ["siding", "spur", "yard"]];
+
+// Bridge casing layers
+export const bridgeCasing = {
+  ...defRail,
+  id: "railway-bridge-casing",
+  filter: [
+    "all",
+    ["==", "brunnel", "bridge"],
+    ["in", "class", "rail", "transit"],
+  ],
+  minzoom: 14,
+  layout: {
+    "line-cap": "butt",
+    "line-join": "bevel",
+    visibility: "visible",
+  },
+  paint: {
+    "line-color": Color.backgroundFill,
+    "line-width": [
+      "interpolate",
+      ["exponential", railExp],
+      ["zoom"],
+      8,
+      [...serviceSelector, 0.8, 1],
+      12,
+      [...serviceSelector, 2.0, 2.5],
+      20,
+      [...serviceSelector, 6.5, 8.5],
+    ],
+    "line-blur": 0.75,
```

Comment:
Against water, the blur ends up looking like a halo rather than a knockout. The rail lines seemed to be slightly more legible to me with a solid edge on the casing. On land, road–rail grade separations are quite subtle. I think we got away with a blur in #203 because the halo was sufficiently thick and contrasted, whereas this casing is quite thin.

Reply:
I think you're right. I went ahead and removed the blur.